cdap-proto/src/main/java/co/cask/cdap/proto/metadata/lineage/FieldLineageDetails.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.metadata.lineage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents the lineage associated with the field of a dataset in detail.
+ * Detail view in the backward direction not only includes the information about the fields
+ * and corresponding datasets which were responsible for generating the field but also includes
+ * the information about operations and programs which were responsible for generating the field.
+ * Similarly detail view in the forward direction not only includes the target fields along with
+ * their dataset information which got generated by this field but also includes the operations
+ * and programs information which were responsible for the transformation.
+ */
+public class FieldLineageDetails {
+  private final List<ProgramFieldOperationInfo> backward;
```

Comment:
thinking about it a little more... perhaps it should be "incoming" and "outgoing", or even something else. forward and backward make less sense to me when I read the API now.

Reply:
perhaps we can mark all the proto classes as @Beta to have a chance to evolve them

Reply:
Marked classes as `@Beta`. Also calling the fields as `incoming` and `outgoing`